docs/java-7-rationale.md

```diff
@@ -0,0 +1,40 @@
+## Rationale for not supporting Java 7
```

Comment:
I like Rationale docs hope we doc out more of them :)

Reply:
I blame your and @adriancole's influence 😁